### PR TITLE
[pkg/ottl] Expose component.TelemetrySettings to OTTL path parsers so that the component's logger is available to contexts

### PR DESCRIPTION
**Pull request by swar8080**
**Description:** <Describe what has changed.>
Passes `component.TelemetrySettings` to all path parsers so that they can pass the component's logger down to current/future contexts' getters/setters

**Link to tracking Issue:** #22076

**Testing:** N/A, i'm assuming a successful compile and tests passing are sufficient

**Documentation:** N/A

### CODE REVIEW

File: pkg/ottl/functions.go [hunk 1]
```diff
@@ -6,13 +6,14 @@ package ottl // import "github.com/open-telemetry/opentelemetry-collector-contri
...
 	"fmt"
+	"go.opentelemetry.io/collector/component"
 	"reflect"
 	"strings"
 
 	"github.com/iancoleman/strcase"
 )
 
-type PathExpressionParser[K any] func(*Path) (GetSetter[K], error)
+type PathExpressionParser[K any] func(component.TelemetrySettings, *Path) (GetSetter[K], error)
```
Comment (TylerHelmuth) [inline pkg/ottl/functions.go:16]: I kinda torn on this. 

On one hand OTTL is specific to the collector so we can expect that users have a TelemetrySettings object to pass in (especially bc the parser has it). 

On the other hand, I like keeping the requirement for a compatible OTTL context low. The desire to have a TelemetrySettings object doesn't have to be built into the public API between the parser and its context. For our specific contexts we could instead handle supplying the TelemetrySettings struct via the contexts themselves.

@evan-bradley what are your thoughts?

Comment (swar8080) [reply pkg/ottl/functions.go:16]: if i'm understanding correctly, the alternative is allowing contexts to opt-in to having a dependency on `component.TelemetrySettings`?

so maybe an approach like this?:

```
type logPathParser struct {
   component.TelemetrySettings
}

func NewParser(functions map[string]ottl.Factory[TransformContext], telemetrySettings component.TelemetrySettings, options ...Option) (ottl.Parser[TransformContext], error) {
	var logPathParser = logPathParser{telemetrySettings}
	p, err := ottl.NewParser[TransformContext](
		functions,
		logPathParser.parsePath,
		telemetrySettings,
		ottl.WithEnumParser[TransformContext](parseEnum),
	)
....
func (lpp *logPathParser) parsePath(_ component.TelemetrySettings, val *ottl.Path) (ottl.GetSetter[TransformContext], error) {
   // access logger through lpp.TelemetrySettings
}

Comment (TylerHelmuth) [reply pkg/ottl/functions.go:16]: Ya that's how I was picturing it if we didn't build it into the ParsePath API

Comment (swar8080) [reply pkg/ottl/functions.go]: okay I switched to that approach

Comment (evan-bradley) [reply pkg/ottl/functions.go]: Sorry for the delay. I much prefer putting TelemetrySettings in a struct field, I think it makes the API cleaner.

**Review by TylerHelmuth — APPROVED** [review]

**Review by evan-bradley — APPROVED** [review]

**Pull request merged**